examples/tests/invalid/inference.test.w

```diff
@@ -96,3 +96,19 @@ let badFunc: inflight (str): void = inflight (arg1: num) => {};
 // ^ Inferred type Array<str> conflicts with already inferred type Array<num>
 };
 
+
+let returnsNumber = () => { return 2; };
+if true {
+  let x: str = returnsNumber();
+//             ^^^^^^^^^^^^^^^ Expected str, got num
+}
+
+let unknownArg = (arg) => {
+//                ^^^ Unable to infer type
+  return arg.get("a");
+//           ^^^ Property not found
+};
+
+if true {
+  unknownArg({ a: true });
+}
```

Comment:
If I understand correctly, this is an only an invalid in the sense that our inference system isn't strong enough to analyze it, not because there's a logical error/bug in the code, right?

If that's the case I don't know if it should be flagged as an invalid case given we might want to support it in the future. (Or maybe we can leave it and link to the issue about implementing better inference?)

Reply:
Yep that's right.

I think there's value in showing this case, since we also want to validate behavior that shouldn't work. This ensures it doesn't change in unexpected ways. I'll add an issue and link to it though.